### ISSUE TITLE
Dashboard filter dropdown not working when linked to question joined table

### ISSUE THREAD

**Issue by gerardo-navarro** [metabase/metabase#12250]
**Describe the bug**
Auto suggestion and dropdown of field filter not working when applied to questions with several join tables.

**To Reproduce**

* Setup a query in the query builder with some join tables.
* Then create a dashboard and add a field filter (field type: entity and field filter type: search or dropdown)
* Apply the added field filter in the dashboard to one of the join tabled, in case of the attached image I applied the field filter to the account manager table
* Once everything is set up, you will notice that the field filter is not auto suggesting the fields anymore

Everything seems to be fine when the field filter is applied to a question with just a single table and no join table.
**Expected behavior**
I would like to see the awesome auto suggestion and dropdown feature that me and my colleagues usually love about the field filters in our dashboards.

**Screenshots**
<img width="1904" alt="image" src="https://user-images.githubusercontent.com/592424/78289241-234fed80-7522-11ea-8ea7-156c7dab6c02.png">

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.1 Safari/605.1.15",
    "vendor": "Apple Computer, Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "1.8.0_242-heroku-b08",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_242-heroku",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "25.242-b08",
    "os.name": "Linux",
    "os.version": "4.4.0-1062-aws",
    "user.language": "en",
    "user.timezone": "Etc/UTC"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "googleanalytics"
    ],
    "hosting-env": "heroku",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.6 (Ubuntu 11.6-1.pgdg16.04+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.35.0",
      "date": "2020-03-31",
      "branch": "master",
      "hash": "3f96c47"
    },
    "settings": {
      "report-timezone": "Europe/Berlin"
    }
  }
}
```
:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by YuriAndreoli** [comment]
Same problem here :(

**Comment by mazameli** [comment]
I reproduced this on 0.35.1:

- Start a new notebook question with the Sample Dataset
- Join Orders to People
- Save the question
- Add it to a dashboard
- Add a dashboard filter and connect it to one of the columns in the joined table, e.g. User State, and save changes
- When you click on the filter, you'll get an input box instead of a dropdown list.

If you remove the join from the question and save it, the filter in the dashboard then gives you a list of options again.

**Comment by freakinruben** [comment]
I'm also experiencing the same problem on 0.35.2

**Comment by ezequielzacca** [comment]
Same problem here, any workaround?

**Comment by maxzheng** [comment]
We have the same issue. A workaround for us is to convert to SQL and use variable field filters. It's a bit more work, but it works.

**Comment by nmcalabroso** [comment]
I'm seeing this issue after upgrading from 0.34.3 to 0.35.4. All our custom questions are working fine by themselves but when viewed in the dashboard with filters, all the the filters that are previously dropdown became text input box.

Not sure yet if this is a backend or frontend problem as I'm seeing this:
<img width="620" alt="Screen Shot 2020-07-20 at 12 20 06 PM" src="https://user-images.githubusercontent.com/5373999/87899305-5abbb900-ca83-11ea-8e8c-7d7c013352d0.png">

I've already checked the Data Model and country is set fine as Category. I've tried re-scaning the field as well.

**Comment by flamber** [comment]
@nmcalabroso Downgrade to 0.34.3 or wait for 0.36.0

**Comment by raphaelmro** [comment]
> I reproduced this on 0.35.1:
> 
> * Start a new notebook question with the Sample Dataset
> * Join Orders to People
> * Save the question
> * Add it to a dashboard
> * Add a dashboard filter and connect it to one of the columns in the joined table, e.g. User State, and save changes
> * When you click on the filter, you'll get an input box instead of a dropdown list.
> 
> If you remove the join from the question and save it, the filter in the dashboard then gives you a list of options again.

It works for me. Does anyone know if this bug is fixed on any recent release?

**Comment by flamber** [comment]
@raphaelmro 
As noted multiple places, this was fixed in 0.36.0. Latest release is 0.39.4: https://github.com/metabase/metabase/releases

There are several other issues open about dropdowns, so have a look at those and upvote the ones you agree with:
https://github.com/metabase/metabase/issues?q=is%3Aissue+is%3Aopen+author%3Aflamber+dropdown+in%3Atitle
